src/ray/gcs/gcs_server/gcs_server.cc

```diff
@@ -372,7 +372,17 @@ void GcsServer::InitGcsResourceManager(const GcsInitData &gcs_init_data) {
           } else {
             raylet_client->GetResourceLoad([this](auto &status, auto &load) {
               if (status.ok()) {
+                // TODO(vitsai): Remove duplicate reporting to GcsResourceManager
+                // after verifying that non-autoscaler paths are taken care of.
+                // Currently, GcsResourceManager aggregates reporting from different
+                // sources at different intervals, leading to an obviously inconsistent
+                // view.
+                //
+                // Once autoscaler is completely moved to the new mode of consistent
+                // per-node reporting, remove this if it is not needed anymore.
                 gcs_resource_manager_->UpdateResourceLoads(load.resources());
+                gcs_autoscaler_state_manager_->UpdateResourceLoadAndUsage(
```

Comment:
Do we also report available/total resources here in `load.resources()`? The RPC name seems to imply it only does for load.

Reply:
i see - it does include. This is a pretty unfortunate in the rpc naming then. Could we also note that down here? E.g. `GetResourceLoad` also includes resource usage information.